### PR TITLE
Docker Compose installation example

### PR DESCRIPTION
**Pull request by gAmUssA**
- initial commit of docker-compose for kong
- updated ignore for jetbrains

### CODE REVIEW

File: kong-docker/dbless/startDbless.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+docker-compose --file docker-compose.yaml up -d
```
Comment (hutchic) [inline kong-docker/dbless/startDbless.sh:3]: I think we have a bias for `Makefile`

File: kong-docker/dbless/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      KONG_PROXY_ERROR_LOG: '/dev/stderr'
+      KONG_ADMIN_ERROR_LOG: '/dev/stderr'
+      KONG_ADMIN_LISTEN: "0.0.0.0:8001, 0.0.0.0:8444 ssl"
+      KONG_DECLARATIVE_CONFIG: "/opt/kong/kong.yaml"
+    command: "kong start"
+    ports:
+      - "8000:8000"
+      - "8443:8443"
+      - "8001:8001"
+      - "8444:8444"
```
Comment (hutchic) [inline kong-docker/dbless/docker-compose.yaml:23]: we try to encourage not opening the admin API port unprotected to the world at large. Doing it as follows makes it only open to localhost though this won't work if you're running the docker-compose somewhere remote obviously

```suggestion
      - "127.0.0.1:8001:8001/tcp"
      - "127.0.0.1:8444:8444/tcp"
```

**Review by hutchic — CHANGES_REQUESTED** [review]
The delta between the two docker-compose files looks to be
```
      KONG_DATABASE: 'off'
```

I'd suggest use a single docker-compose with environment variables w\ sensible defaults. Adjust the Makefile to account for the two options

docker-compose.yaml
```
      KONG_DATABASE: '"${KONG_DOCKER_DATABASE:-postgres}"'
```

Makefile
```
kong-postgres:
    docker-compose up -d

kong-dbless:
   KONG_DOCKER_DATABASE=off docker-compose up -d
```

Also we could largely capture these changes in https://github.com/Kong/docker-kong/tree/master/compose no?

File: kong-docker/dbless/startDbless.sh (hunk 1 above)
Comment (gAmUssA) [reply kong-docker/dbless/startDbless.sh]: noted. added makefile

**Comment by gAmUssA** [comment]
We will keep this one for demo purposes, and point users to production-ready in Kong-docker repo @hutchic

**Pull request merged**